src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs

```diff
@@ -257,9 +257,15 @@ public MetadataReference WithProperties(MetadataReferenceProperties properties)
         /// Reusing <see cref="AssemblyMetadata"/> object allows for sharing data accross these references.
         /// </para>
         /// </remarks>
+        [Obsolete("The API is obsolete, use CreateFromFile(assembly.Location) instead", error: true)]
```

Comment:
Please also add `[EditorBrowsable(EditorBrowsableState.Never)]` so that IntelliSense doesn't suggest this method.